webdriver/tests/bidi/script/call_function/sandbox.py

```diff
@@ -0,0 +1,179 @@
+        await_promise=True,
+    )
+    assert result_in_window == {"type": "undefined"}
+
+
+@pytest.mark.asyncio
+async def test_sandbox_with_empty_name(bidi_session, new_tab):
+    # BiDi specification doesn't have restrictions of a sandbox name,
+    # that's why we want to make sure that it works with an empty name
+    await bidi_session.script.call_function(
+        function_declaration="() => window.foo = 'bar'",
+        target=ContextTarget(new_tab["context"], ""),
+        await_promise=True,
+    )
+
+    # Make sure that we can find the sandbox with the empty name
+    result = await bidi_session.script.call_function(
+        function_declaration="() => window.foo",
+        target=ContextTarget(new_tab["context"], ""),
+        await_promise=True,
+    )
+    assert result == {"type": "string", "value": "bar"}
+
+    # Make sure that changes didn't leak from sandbox
+    result = await bidi_session.script.evaluate(
+        expression="window.foo",
+        target=ContextTarget(new_tab["context"]),
+        await_promise=True,
+    )
+    assert result == {"type": "undefined"}
+
+
+@pytest.mark.asyncio
+async def test_switch_sandboxes(bidi_session, new_tab):
+    # Test that sandboxes are retained when switching between them
+    await bidi_session.script.call_function(
+        function_declaration="() => { window.foo = 1; }",
+        target=ContextTarget(new_tab["context"], "sandbox_1"),
+        await_promise=True,
+    )
+    await bidi_session.script.call_function(
+        function_declaration="() => { window.foo = 2; }",
+        target=ContextTarget(new_tab["context"], "sandbox_2"),
+        await_promise=True,
+    )
+
+    result_in_sandbox_1 = await bidi_session.script.call_function(
+        function_declaration="() => window.foo",
+        target=ContextTarget(new_tab["context"], "sandbox_1"),
+        await_promise=True,
+    )
+    assert result_in_sandbox_1 == {"type": "number", "value": 1}
+
+    result_in_sandbox_2 = await bidi_session.script.call_function(
+        function_declaration="() => window.foo",
+        target=ContextTarget(new_tab["context"], "sandbox_2"),
+        await_promise=True,
+    )
+    assert result_in_sandbox_2 == {"type": "number", "value": 2}
+
+
+@pytest.mark.asyncio
+async def test_sandbox_with_side_effects(bidi_session, new_tab):
+    # Make sure changing the node in sandbox will affect the other sandbox as well
+    await bidi_session.script.call_function(
+        function_declaration="() => document.querySelector('body').textContent = 'foo'",
+        target=ContextTarget(new_tab["context"], "sandbox_1"),
+        await_promise=True,
+    )
+
+    result_in_sandbox_1 = await bidi_session.script.call_function(
+        function_declaration="() => document.querySelector('body').textContent",
+        target=ContextTarget(new_tab["context"], "sandbox_1"),
+        await_promise=True,
+    )
+    assert result_in_sandbox_1 == {"type": "string", "value": "foo"}
```

Comment:
nit: feel free to set it as a constant (`expected_value`) at the top and re-use it for sandbox2 as well.